packages/@aws-cdk/cdk/lib/cloudformation/output.ts

```diff
@@ -124,6 +132,19 @@ export class Output extends StackElement {
   public get ref(): string {
     throw new Error('Outputs cannot be referenced');
   }
+
+  /**
+   * Automatically determine an output name for use with FnImportValue
+   *
+   * This gets called in case the user hasn't specified an export name but is
+   * taking an action that requires exporting. We namespace with the stack name
+   * to reduce chances of collissions between CDK apps.
+   */
+  private uniqueOutputName() {
+    // prefix export name with stack name since exports are global within account + region.
+    const stackName = require('./stack').Stack.find(this).name;
```

Comment:
I do not like this inline `require` call. Why not import `Stack` normally?

Reply:
Cyclic dependencies. The core library is rife with them. :(